homeassistant/components/apsystemsapi_local/sensor.py

```diff
@@ -0,0 +1,175 @@
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL,
+        value_fn=lambda c: c.te2,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="today_production",
+        translation_key="today_production",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda c: c.e1 + c.e2,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="today_production_p1",
+        translation_key="today_production_p1",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda c: c.e1,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="today_production_p2",
+        translation_key="today_production_p2",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda c: c.e2,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: config_entries.ConfigEntry,
+    add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the sensor platform."""
+    config = hass.data[DOMAIN][config_entry.entry_id]
+    coordinator = config["COORDINATOR"]
+    device_name = config[CONF_NAME]
+
+    add_entities(
+        ApSystemsSensorWithDescription(coordinator, desc, device_name)
+        for desc in SENSORS
+    )
+
+
+class ApSystemsSensorWithDescription(CoordinatorEntity, SensorEntity):
+    """Base sensor to be used with description."""
+
+    entity_description: ApsystemsLocalApiSensorDescription
+
+    def __init__(
+        self,
+        coordinator: ApSystemsDataCoordinator,
+        entity_description: ApsystemsLocalApiSensorDescription,
+        device_name: str,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self._state: int | float | None = None
+        self.entity_description = entity_description
+        self._device_name = device_name
+        self._attr_unique_id = f"apsystemsapi_{device_name}_{entity_description.key}"
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Get the DeviceInfo."""
+        return DeviceInfo(
+            identifiers={("apsystemsapi_local", self._device_name)},
+            name=self._device_name,
+            manufacturer="APsystems",
+            model="EZ1-M",
+        )
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        if self.coordinator.data is not None:
+            self._state = self.entity_description.value_fn(self.coordinator.data)
+        self.async_write_ha_state()
+
+    @property  # type: ignore[misc]
+    def state(self) -> float | None:
```

Comment:
Don't override state, that's managed by the sensor class.

```suggestion
    def native_value(self) -> float | None:
```